cpu/stm32/Makefile

```diff
@@ -3,6 +3,9 @@ MODULE = cpu
 DIRS = $(RIOTCPU)/cortexm_common periph stmclk vectors
 
 ifneq (,$(filter bootloader_stm32,$(USEMODULE)))
+  ifeq ($(CPU_FAM), mp1)
+    $(error STM32MP1 family has no flash and thus no bootloader can be used)
+  endif
```

Comment:
Just don't declare the bootloader_stm32 feature as provided on mp1 (in Makefile.features) and you can drop this.